migration/sql-files/021-organizations-list-create.sql

```diff
@@ -0,0 +1,12 @@
+ALTER TABLE identity_role drop column identity_role_id;
+ALTER TABLE identity_role ADD COLUMN identity_role_id uuid DEFAULT uuid_generate_v4();
+ALTER TABLE identity_role ADD CONSTRAINT pk_identity_role PRIMARY KEY (identity_role_id);
+CREATE UNIQUE INDEX uq_identity_role_identity_role_resource ON identity_role (identity_id, resource_id, role_id);
+
+CREATE UNIQUE INDEX uq_role_resource_type_name ON role (resource_type_id, name);
+
+INSERT INTO resource_type (name) VALUES ('identity/organization');
+
+INSERT INTO role (role_id, resource_type_id, name) select uuid_generate_v4(), resource_type_id, 'owner' from resource_type where name = 'identity/organization';
```

Comment:
Can we add tests in some form to test the `identity/organization` insertion in the relevant table(s) ?

```
INSERT INTO resource_type (name) VALUES ('identity/organization');
INSERT INTO role (role_id, resource_type_id, name) select uuid_generate_v4(), resource_type_id, 'owner' from resource_type where name = 'identity/organization';
```

Reply:
We have these covered already - see:

[](https://github.com/fabric8-services/fabric8-auth/pull/317/files#diff-3c2879c729984887477d67ffd0960ffaR19)
[](https://github.com/fabric8-services/fabric8-auth/pull/317/files#diff-3c2879c729984887477d67ffd0960ffaR35)

Also a similar test exists for roles.

Reply:
The migration tests are run in a database called `test` , which is not the same database where our other tests run. So effectively the above lines don't really cover that :) 
This `test` database is one which is created only to test out migration SQLs.
So in the `testMigration21` you could have something like this to test out whether the migration SQLs actually produce the intended outcome.
```
        rows, err := sqlDB.Query("SELECT name FROM resource_type")
	if err != nil {
		t.Fatal(err)
	}
	defer rows.Close()
	for rows.Next() {
                var resourceTypeName string
		err = rows.Scan(&resourceTypeName)
               // assert resourceTypeName ....
               // .....
	}
```

Reply:
No problem, I can add a check here for new types.